pkg/gameserversets/controller.go

```diff
@@ -440,11 +440,12 @@ func (c *Controller) deleteGameServers(gsSet *v1alpha1.GameServerSet, toDelete [
 	c.logger.WithField("diff", len(toDelete)).WithField("gameserverset", gsSet.ObjectMeta.Name).Info("Deleting gameservers")
 
 	return parallelize(gameServerListToChannel(toDelete), maxDeletionParallelism, func(gs *v1alpha1.GameServer) error {
-		c.allocationMutex.Lock()
-		err := c.gameServerGetter.GameServers(gs.Namespace).Delete(gs.ObjectMeta.Name, nil)
-		c.allocationMutex.Unlock()
+		// We should not delete the gameservers directly buy set their state to shutdown and let the gameserver controller to delete
```

Comment:
perhaps explain the Allocate/Delete race this is resolving

Reply:
GS Allocate and Delete activities were being synced across all the controllers. This is/will adversely impacting the performance. Since the delete is only happening here for the GS with Shutdown state, those can never be allocated hence no need to sync.